src/cargo/core/compiler/build_config.rs

```diff
@@ -97,11 +97,6 @@ impl BuildConfig {
             },
         };
 
-        if gctx.cli_unstable().build_std.is_some() && requested_kinds[0].is_host() {
```

Comment:
Should we add some `build-std` tests for relying on host target?

Reply:
This might not be easy to set up though. Maybe we can only do this on certain platforms like x86_64 linux.

Reply:
@ehuss I meant this but now it seems to be a separate issue so you're right.

https://github.com/rust-lang/cargo/blob/a7c4206e1719d57cb7af323904e139067b95b8fa/tests/build-std/main.rs#L235-L240